core/server/update-check.js

```diff
@@ -142,78 +164,123 @@ function updateCheckRequest() {
+        var update  = {},
+            version = response.settings[0];
 
         // Version 0.4 used boolean to indicate the need for an update. This special case is
         // translated to the version string.
         // TODO: remove in future version.
-        if (display.value === 'false' || display.value === 'true' || display.value === '1' || display.value === '0') {
-            display.value = '0.4.0';
+        if (version.value === 'false' || version.value === 'true' || version.value === '1' || version.value === '0') {
+            version.value = '0.4.0';
         }
 
-        if (display && display.value && currentVersion && semver.gt(display.value, currentVersion)) {
-            return display.value;
+        // available: true if a new version is available.
+        update = {
+            available: version && version.value && currentVersion && semver.gt(version.value, currentVersion),
+            version: version.value
+        };
+
+        return update;
+    }).then(function then(update) {
+        var updateNotification = null,
+            getStoredNotifications = api.notifications.browse({context: {internal: true}}),
+            getSeenNotifications   = api.settings.read({context: {internal: true}, key: 'seenNotifications'});
+
+        // If a new version is available, construct a notification
+        if (update.available) {
+            updateNotification = {
+                type: 'upgrade',
+                location: 'settings-about-upgrade',
+                dismissible: false,
+                status: 'alert',
+                message: i18n.t('notices.controllers.newVersionAvailable', {
+                    version: update.version,
+                    link: '<a href="http://support.ghost.org/how-to-upgrade/" target="_blank">Click here</a>'
+                })
+            };
         }
 
-        return false;
+        return [
+            updateNotification,     // upgrade availability notification
+            getStoredNotifications, // currently available notifications
+            getSeenNotifications    // UUIDs of seen notifications
+        ];
+    }).spread(function spread(updateNotification, stored, seen) {
+        var notifications     = [],
+            seenNotifications = safeParseJson(seen.settings[0].value);
+
+        // Remove seen notifications and duplicates before saving all
+        notifications = _.chain(stored.notifications)
+        .filter(function filter(n) {
+            var wasSeen = _.has(n, 'uuid') && _.contains(seenNotifications, n.uuid);
+            return !wasSeen;
+        })
+        .uniqBy('message')
+        .value();
+
+        // If update notification exists and it's not already in the store, add...
+        if (updateNotification && !_.some(stored.notifications, {message: updateNotification.message})) {
+            notifications.push(updateNotification);
+        }
+
+        return api.notifications.add({notifications: notifications}, {context: {internal: true}});
```

Comment:
This throws errors for me because it tries to add an empty notifications array if there aren't any.